packages/runtimes/js/src/JSRuntime.js

```diff
@@ -475,6 +475,24 @@ function getLoaderRuntime({
     loaderCode = `(${loaderCode})`;
   }
 
+  if (needsEsmLoadPrelude && options.featureFlags.importRetry) {
+    const assetId = bundleGraph.getAssetById(bundleGroup.entryAssetId);
+    loaderCode = `
+      Object.defineProperty(module, 'exports', { get: () => {
```

Comment:
Is there a reason for this module definition stuff?

Reply:
At the call site we call `parcelRequire` directly, expecting it to return a promise:

```javascript
const abc = parcelRequire("abc")
```

The value resolved by the `Promise` is cached so subsequent calls will not rerun the constructor. Using a getter ensures the promise is recreated - essentially hacking it to be 

```javascript
const abc = parcelRequire("abc")()
```

That said, I should probably only remove the cache entry if there is a failure in a request. Will try to do that